src/__tests__/useMutationObserver.spec.tsx

```diff
@@ -0,0 +1,156 @@
+        const todo = document.createElement("li");
+        id += 1;
+        todo.textContent = `todo ${id}`;
+        ref.current?.appendChild(todo);
+      };
+
+      useMutationObserver(ref, (mutations) => {
+        for (const mutation of mutations) {
+          if (mutation.type === "childList" && mutation.addedNodes) {
+            mutation.addedNodes.forEach((node) => {
+              console.log(`${node.textContent} has been added to todo list.`);
+            });
+          }
+        }
+      });
+
+      return (
+        <div>
+          <ul ref={ref}>
+            <li>todo {id}</li>
+          </ul>
+          <button onClick={addTodo} type="button">
+            Add Todo
+          </button>
+        </div>
+      );
+    }
+
+    render(<TestComp />);
+    const button = screen.getByText(/add todo/i);
+    fireEvent.click(button);
+    expect(screen.getAllByRole("listitem")).toHaveLength(2);
+  });
+
+  test("should observe the same DOM node multiple times when provide multiple listeners", async () => {
+    expect.hasAssertions();
+    const listener1 = jest.fn();
+    const listener2 = jest.fn();
+    function TestComp() {
+      const ref = useRef<HTMLDivElement>(null);
+      useMutationObserver(ref, listener1);
+      useMutationObserver(ref, listener2);
+
+      return (
+        <div>
+          <div ref={ref}>status: coding</div>
+          <button
+            onClick={() => {
+              if (ref.current) {
+                ref.current.textContent = "status: Gaming";
+              }
+            }}
+            type="button"
+          >
+            Start to play game!
+          </button>
+        </div>
+      );
+    }
+    render(<TestComp />);
+    const button = screen.getByText(/start to play game!/i);
+    fireEvent.click(button);
+    await waitFor(() => {
+      expect(listener1).toHaveBeenCalled();
+      expect(listener2).toHaveBeenCalled();
+    });
+  });
+
+  test("should observe the DOM node multiple times even if the listener is same", async () => {
+    expect.hasAssertions();
+    let calls = 0;
+    function TestComp() {
+      const ref = useRef<HTMLDivElement>(null);
+
+      const listener = () => {
+        calls++;
+      };
+
+      useMutationObserver(ref, listener);
+      useMutationObserver(ref, listener);
+
+      return (
+        <div>
+          <div ref={ref}>status: coding</div>
+          <button
+            onClick={() => {
+              if (ref.current) {
+                ref.current.textContent = "status: Gaming";
+              }
+            }}
+            type="button"
+          >
+            Start to play game!
+          </button>
+        </div>
+      );
+    }
+    render(<TestComp />);
+    const button = screen.getByText(/start to play game!/i);
+    fireEvent.click(button);
+    await waitFor(() => {
+      expect(calls).toBe(2);
+    });
+  });
+
+  test("should stop the MutationObserver instance from receiving further notifications", async () => {
```

Comment:
We could name this test slightly better, but this doesn't need to happen in this PR. Looks great overall!

Reply:
My English is poor. Some statements may not conform to common English usage. Help me fix it. Thanks.

Reply:
Don't worry about it! English is not the most important thing here.